[Package]: Raylib

### Package description

raylib is a simple and easy-to-use library to enjoy videogames programming.

raylib is highly inspired by Borland BGI graphics lib and by XNA framework and it's specially well suited for prototyping, tooling, graphical applications, embedded systems and education.

### Home page URL

http://www.raylib.com/

### Source code URL

https://github.com/raysan5/raylib

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/x11-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

I tried to compile it on Termux but got this error:
```
raylib/build on  master via △ v3.21.1 took 13s
❯ make -j4                                                                      11:04:22
[  1%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/context.c.o
[  1%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/init.c.o
[  2%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/input.c.o
[  2%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/monitor.c.o
[  2%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/vulkan.c.o
[  3%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/window.c.o
[  3%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/x11_init.c.o
[  3%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/x11_monitor.c.o
[  4%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/x11_window.c.o
[  4%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/xkb_unicode.c.o
[  5%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/posix_time.c.o
[  5%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/posix_thread.c.o
[  5%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/glx_context.c.o
[  6%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/egl_context.c.o
[  6%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/osmesa_context.c.o
[  6%] Building C object raylib/external/glfw/src/CMakeFiles/glfw.dir/null_joystick.c.o
[  7%] Linking C static library libglfw3.a
[  7%] Built target glfw
[  7%] Building C object raylib/CMakeFiles/raylib.dir/core.c.o
[  8%] Building C object raylib/CMakeFiles/raylib.dir/models.c.o
[  8%] Building C object raylib/CMakeFiles/raylib.dir/shapes.c.o
[  9%] Building C object raylib/CMakeFiles/raylib.dir/text.c.o
In file included from /data/data/com.termux/files/home/raylib/src/models.c:81:
/data/data/com.termux/files/home/raylib/src/external/par_shapes.h:1133:32: warning: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648 [-Wimplicit-const-int-float-conversion]
    float r = (float) rand() / RAND_MAX;
                             ~ ^~~~~~~~
/data/data/com.termux/files/usr/include/stdlib.h:133:18: note: expanded from macro 'RAND_MAX'
#define RAND_MAX 0x7fffffff
                 ^~~~~~~~~~
[  9%] Building C object raylib/CMakeFiles/raylib.dir/textures.c.o
[  9%] Building C object raylib/CMakeFiles/raylib.dir/utils.c.o
[ 10%] Building C object raylib/CMakeFiles/raylib.dir/raudio.c.o
1 warning generated.
[ 10%] Linking C static library libraylib.a
[ 10%] Built target raylib
[ 11%] Building C object examples/CMakeFiles/textures_mouse_painting.dir/textures/textures_mouse_painting.c.o
[ 11%] Building C object examples/CMakeFiles/textures_to_image.dir/textures/textures_to_image.c.o
[ 11%] Building C object examples/CMakeFiles/textures_sprite_explosion.dir/textures/textures_sprite_explosion.c.o
[ 11%] Building C object examples/CMakeFiles/textures_npatch_drawing.dir/textures/textures_npatch_drawing.c.o
[ 12%] Linking C executable textures_to_image
[ 12%] Linking C executable textures_sprite_explosion
[ 13%] Linking C executable textures_npatch_drawing
[ 14%] Linking C executable textures_mouse_painting
/data/data/com.termux/files/usr/bin/ld: /data/data/com.termux/files/usr/bin/ld: ../raylib/libraylib.a(x11_window.c.o): in function `_glfwPlatformPollEvents':
/data/data/com.termux/files/home/raylib/src/external/glfw/src/x11_window.c:2784: undefined reference to `_glfwDetectJoystickConnectionLinux'
../raylib/libraylib.a(x11_window.c.o): in function `_glfwPlatformPollEvents':
/data/data/com.termux/files/home/raylib/src/external/glfw/src/x11_window.c:2784: undefined reference to `_glfwDetectJoystickConnectionLinux'
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
/data/data/com.termux/files/usr/bin/ld: ../raylib/libraylib.a(x11_window.c.o): in function `_glfwPlatformPollEvents':
/data/data/com.termux/files/home/raylib/src/external/glfw/src/x11_window.c:2784: undefined reference to `_glfwDetectJoystickConnectionLinux'
make[2]: *** [examples/CMakeFiles/textures_to_image.dir/build.make:102: examples/textures_to_image] Error 1
make[1]: *** [CMakeFiles/Makefile2:441: examples/CMakeFiles/textures_to_image.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: *** [examples/CMakeFiles/textures_npatch_drawing.dir/build.make:102: examples/textures_npatch_drawing] Error 1
make[1]: *** [CMakeFiles/Makefile2:495: examples/CMakeFiles/textures_npatch_drawing.dir/all] Error 2
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [examples/CMakeFiles/textures_sprite_explosion.dir/build.make:102: examples/textures_sprite_explosion] Error 1
make[1]: *** [CMakeFiles/Makefile2:468: examples/CMakeFiles/textures_sprite_explosion.dir/all] Error 2
/data/data/com.termux/files/usr/bin/ld: ../raylib/libraylib.a(x11_window.c.o): in function `_glfwPlatformPollEvents':
/data/data/com.termux/files/home/raylib/src/external/glfw/src/x11_window.c:2784: undefined reference to `_glfwDetectJoystickConnectionLinux'
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [examples/CMakeFiles/textures_mouse_painting.dir/build.make:102: examples/textures_mouse_painting] Error 1
make[1]: *** [CMakeFiles/Makefile2:522: examples/CMakeFiles/textures_mouse_painting.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

raylib/build on  master via △ v3.21.1 took 2m56s
❯                                                                               11:07:18
```

--- a/x11-packages/raylib/build.sh
+++ b/x11-packages/raylib/build.sh
@@ -1,0 +1,30 @@
+TERMUX_PKG_HOMEPAGE=https://www.raylib.com/
+TERMUX_PKG_DESCRIPTION="A simple and easy-to-use library to enjoy videogames programming"
+TERMUX_PKG_LICENSE="ZLIB"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=4.0.0
+TERMUX_PKG_SRCURL=https://github.com/raysan5/raylib/archive/refs/tags/${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=11f6087dc7bedf9efb3f69c0c872f637e421d914e5ecea99bbe7781f173dc38c
+TERMUX_PKG_DEPENDS="glfw"
+TERMUX_PKG_BUILD_DEPENDS="mesa"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DPLATFORM=Desktop
+-DBUILD_EXAMPLES=OFF
+-DBUILD_SHARED_LIBS=ON
+-DUSE_EXTERNAL_GLFW=ON
+"
+
+termux_step_pre_configure() {
+	_NEED_DUMMY_LIBPTHREAD_A=
+	_LIBPTHREAD_A=$TERMUX_PREFIX/lib/libpthread.a
+	if [ ! -e $_LIBPTHREAD_A ]; then
+		_NEED_DUMMY_LIBPTHREAD_A=true
+		echo '!<arch>' > $_LIBPTHREAD_A
+	fi
+}
+
+termux_step_post_make_install() {
+	if [ $_NEED_DUMMY_LIBPTHREAD_A ]; then
+		rm -f $_LIBPTHREAD_A
+	fi
+}

